### PR TITLE
bpo-35133: Fix mistakes when concatenate string literals on different lines.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Two kind of mistakes:

1. Missed space. After concatenating there is no space between words.

2. Missed comma. Causes unintentional concatenating in a list of strings.

https://bugs.python.org/issue35133

### CODE REVIEW

File: Lib/idlelib/idle_test/htest.py
Comment (ericvsmith) [inline Lib/idlelib/idle_test/htest.py]: I realize the period after `console` was already on this line, but does it make any sense? I think it should be deleted, and the new end of this line should be `console "`.

File: Lib/test/test_email/test__header_value_parser.py
Comment (ericvsmith) [inline Lib/test/test_email/test__header_value_parser.py]: What about the missing space before `John` on this and subsequent lines? I realize it's not the issue you're solving, but since you're adding spaces after commas, do you need one here, too? Or maybe the point of this test is to not have spaces after commas? I haven't looked at it closely yet.

File: Lib/test/test_http_cookiejar.py
Comment (ericvsmith) [inline Lib/test/test_http_cookiejar.py:177]: This looks suspicious to me. Why is the line starting with `1980-01-01` dropped?

**Review by ericvsmith — COMMENTED** [review]
I have a few comments, the others look good. Although this whole change just points out a pitfall of having max. line length constraints: errors like the ones you're fixing are pretty common, and hard to spot.

File: Lib/test/test_email/test__header_value_parser.py
Comment (serhiy-storchaka) [reply Lib/test/test_email/test__header_value_parser.py]: These tests is the only thing about which I am unsure.

File: Lib/test/test_http_cookiejar.py
Comment (serhiy-storchaka) [reply Lib/test/test_http_cookiejar.py:177]: Because it is in valid format. Test fails because `iso2time()` returns not None. I don't know what invalid example was meant, so I have deleted it.

File: Lib/unittest/mock.py
Comment (tirkarthi) [inline Lib/unittest/mock.py:1775]: Good catch! This would have been caught only if the test had used `__prepare__` or `__instancecheck__` since importing `_unsupported_magics` for test iteration would have still passed at https://github.com/python/cpython/blob/bf46a09dec372b85846216bd692d648dac08ac36/Lib/unittest/test/testmock/testmagicmethods.py#L356 ;)

File: Lib/email/_header_value_parser.py
Comment (terryjreedy) [inline Lib/email/_header_value_parser.py]:
```
        section.defects.append(errors.InvalidHeaderError("section number "
            "has an invalid leading 0"))
```
could evade the issue and be a bit more readable with
```
        section.defects.append(errors.InvalidHeaderError(
            "section number has an invalid leading 0"))
```

File: Lib/idlelib/idle_test/htest.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/htest.py:123]: Reading this, I think putting '\n' anywhere other that at the end of a physical line makes the split msg literal less readable.  I will revise this and a few others in a different PR.

File: Lib/idlelib/idle_test/htest.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/htest.py:339]: Yes, '.' was supposed to be ' '.

File: Lib/test/test_email/test__header_value_parser.py
Comment (terryjreedy) [inline Lib/test/test_email/test__header_value_parser.py]: This fix is marginal in that each concatenated string literal is structured data that is an argument to parser.get_group.    Since the test passes, the data grammar makes the added space at most optional, and its absence is a style issue.  I think the email team should decide on this.

This is quite different from adding needed spaces in text representing English.  Misspelling 'console or', for instance, as 'console.or' (or 'consoleor', if I had just omitted the space) is simply wrong.

File: Lib/test/test_email/test__header_value_parser.py
Comment (terryjreedy) [reply Lib/test/test_email/test__header_value_parser.py]: The email team should look and decide.   This again, is structured data, not English prose.

Comment (warsaw) [reply Lib/test/test_email/test__header_value_parser.py]: I'm not 100% positive, but I think it's okay in this case.  I would wait for comment from @bitdancer though.

File: Lib/test/test_http_cookiejar.py
Comment (terryjreedy) [inline Lib/test/test_http_cookiejar.py:181]: This seems like a different issue.  Should not the \n and the 2nd line of the original assert failure message just be dropped?

**Review by terryjreedy — COMMENTED** [review]
The fixes for list errors and English misspellings (other than the assert failure message) look correct.

I think an email person (@bitdancer ) should look at the email test data formatting -- or omit the changes.

File: Lib/idlelib/idle_test/htest.py
Comment (terryjreedy) [reply Lib/idlelib/idle_test/htest.py]: Right.

File: Lib/test/test_http_cookiejar.py
Comment (ericvsmith) [reply Lib/test/test_http_cookiejar.py:177]: Ah. Okay.

File: Lib/test/test_email/test__header_value_parser.py
Comment (bitdancer) [reply Lib/test/test_email/test__header_value_parser.py]: Given that I included the space in the last line in each set, I'm pretty sure the omission was intentional.  I think it was making sure the ( after the comma got parsed correctly.  It would be better to test such things separately, but there are already so many tests that I started to throw the small variations into other tests...

File: Lib/test/test_email/test__header_value_parser.py
Comment (bitdancer) [reply Lib/test/test_email/test__header_value_parser.py]: I think you should leave all the email tests alone.  Having a mixture of spaces after commas and no spaces after commas is good thing in these tests, and although I don't remember for sure, most likely intentional.

**Review by bitdancer — DISMISSED** [review]

File: Lib/test/test_http_cookiejar.py
Comment (serhiy-storchaka) [reply Lib/test/test_http_cookiejar.py:181]: "is not None" is already output in the error report, as well as the result of iso2time():
```
AssertionError: 315622800 is not None : iso2time(1980-01-01 00:00:00 -2500) is not None
iso2time(test) 315622800
```
I have removed some redundancy:
```
AssertionError: 315622800 is not None : iso2time('1980-01-01 00:00:00 -2500')
```

**Comment by serhiy-storchaka** [comment]
I have made the requested changes; please review again.

**Review by asottile — APPROVED** [review]
this is amazing, how did you find all of these? is there a tool or was it through some `grep`?

**Comment by serhiy-storchaka** [comment]
There are tools for multiline searching, but it was easier to me to write a simple Python script.

File: Lib/test/test_email/test__header_value_parser.py
Comment (terryjreedy) [reply Lib/test/test_email/test__header_value_parser.py]: Changes to test_email were removed in commit 44c6760

**Review by terryjreedy — APPROVED** [review]
The changes requested by myself and David Murray have be made.

**Comment by terryjreedy** [comment]
@bitdancer Since you have not re-reviewed since 44c6760e2b65068066a3d8c7a2d7073fe9374ae8 removed the changes to test_email, 4 days ago, I guessed that you must be busy.  So your request was easy to confirm, I took the initiative of doing so in your behalf and dismissing your negative review, and giving Serhiy an OK to merge.  If this annoys you, let me know that I should not repeat.

**Comment by serhiy-storchaka** [comment]
Thank you all for your reviews!

**Pull request merged**
